lib/misc/cells.js

```diff
@@ -0,0 +1,64 @@
+(function() {
+  var scopes, values;
+
+  scopes = require('./scopes');
+
+  values = require('underscore-plus').values;
+
+  module.exports = {
+    getRange: function(ed) {
+      var buffer, cursor, end, regex, regexString, start;
+      buffer = ed.getBuffer();
+      start = buffer.getFirstPosition();
+      end = buffer.getEndPosition();
+      regexString = atom.config.get('julia-client.cellDelimiter');
+      regex = new RegExp(regexString);
+      cursor = ed.getCursorBufferPosition();
+      if (cursor.row > 0) {
+        buffer.backwardsScanInRange(regex, [start, cursor], function(arg) {
+          var range;
+          range = arg.range;
+          return start = range.start;
+        });
+      }
+      buffer.scanInRange(regex, [cursor, end], function(arg) {
+        var range;
+        range = arg.range;
+        return end = range.start;
```

Comment:
That doesn't really work either, because that way you can't add new lines to the end of an already eval'ed cell without having the output somewhere in the middle (same behaviour you already get at the end of a file right now -- should probably also be changed so that the result is always below the last line in the editor or something. Not critical though, imho).